_posts/2025-04-16-tha-various-tiers-of-tiered-storage.md

```diff
@@ -0,0 +1,454 @@
+-rw-r--r--. 1 1000740000 1000740000 10484635 Apr 16 20:24 00000000000002370608.log
+-rw-r--r--. 1 1000740000 1000740000      240 Apr 16 20:24 00000000000002370608.snapshot
+-rw-r--r--. 1 1000740000 1000740000     1248 Apr 16 20:24 00000000000002370608.timeindex
+-rw-r--r--. 1 1000740000 1000740000 10485760 Apr 16 20:24 00000000000002380960.index
+-rw-r--r--. 1 1000740000 1000740000  1912190 Apr 16 20:24 00000000000002380960.log
+-rw-r--r--. 1 1000740000 1000740000      240 Apr 16 20:24 00000000000002380960.snapshot
+-rw-r--r--. 1 1000740000 1000740000 10485756 Apr 16 20:24 00000000000002380960.timeindex
+-rw-r--r--. 1 1000740000 1000740000        8 Apr 16 20:16 leader-epoch-checkpoint
+-rw-r--r--. 1 1000740000 1000740000       43 Apr 16 20:16 partition.metadata
+```
+
+So, where did the old segments go?
+They were offloaded to the remote storage tier.
+And because we use NFS, we can easily verify that with the `ls` command again.
+Inside our NFS volume at the `/mnt/tiered-storage/` path, you will see a subdirectory for each topic that is using tiered storage.
+And inside the topic subdirectory another subdirectory for each partition.
+And inside that, you will find all the segments that were offloaded to the remote storage tier:
+
+```
+kubectl exec -ti my-cluster-broker-0 -- ls -l /mnt/tiered-storage/tiered-storage-test-ZLI5GN1xR1aOqKnwKu5NOQ/0
+total 5710364
+-rw-r--r--. 1 1000740000 1000740000     6808 Apr 16 20:20 00000000000000000000-eKFr45f2Sca1O0kauyoFBw.indexes
+-rw-r--r--. 1 1000740000 1000740000 10484635 Apr 16 20:20 00000000000000000000-eKFr45f2Sca1O0kauyoFBw.log
+-rw-r--r--. 1 1000740000 1000740000      736 Apr 16 20:20 00000000000000000000-eKFr45f2Sca1O0kauyoFBw.rsm-manifest
+-rw-r--r--. 1 1000740000 1000740000     6172 Apr 16 20:20 00000000000000010352-DmFiyQe8TpOzrpUjPF3rMQ.indexes
+-rw-r--r--. 1 1000740000 1000740000 10484650 Apr 16 20:20 00000000000000010352-DmFiyQe8TpOzrpUjPF3rMQ.log
+-rw-r--r--. 1 1000740000 1000740000      743 Apr 16 20:20 00000000000000010352-DmFiyQe8TpOzrpUjPF3rMQ.rsm-manifest
+-rw-r--r--. 1 1000740000 1000740000     6520 Apr 16 20:20 00000000000000020704-WpjsPsRHTlivhNhxKOpGMA.indexes
+-rw-r--r--. 1 1000740000 1000740000 10484680 Apr 16 20:20 00000000000000020704-WpjsPsRHTlivhNhxKOpGMA.log
+-rw-r--r--. 1 1000740000 1000740000      744 Apr 16 20:20 00000000000000020704-WpjsPsRHTlivhNhxKOpGMA.rsm-manifest
+...
+```
+
+Let's also try to consume the messages to make sure that the broker will correctly retrieve the data from the remote tier and deliver them to the consumer.
+We can use the following command to show us the timestamp of the oldest messages in our topic:
+
+```
+$ kubectl run kafka-consumer -ti --image=quay.io/strimzi/kafka:0.45.0-kafka-3.9.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic tiered-storage-test --from-beginning --max-messages 10 --property print.timestamp=true --property print.value=false
```

Comment:
While using timestamp can implicitly show it's the earliest records we sent, I personally like to use `--property print.offset=true` to show the offset (i.e. offset: 0) to show explicitly it's the offset 0 record. But both ways work for sure. FYI.

Reply:
I updated it and added the offsets to it as well. Thanks.